src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -135,7 +135,7 @@ bool HasConditionalDirectives()
 
                     static bool IsSingleThrowNotImplementedOperation(IOperation firstBlock)
                     {
-                        var compilation = firstBlock.SemanticModel.Compilation;
+                        var compilation = firstBlock.SemanticModel!.Compilation;
```

Comment:
❔ In what cases is this null?

Reply:
When you're operating on operations that come from a CFG. In the original nodes, it's never null.

Reply:
Beyond docs (which we have on the property) I don't know of a good way to annotate that other than `?` and suppress/assert the warnings away.

Reply:
The only valid use of SemanticModel is when it does NOT come from CFG, that is, it's still a bug if you check for null in that case, because it's always null. I think we should return `SemanticModel` and throw if it's not available.

Reply:
> I think we should return SemanticModel and throw if it's not available.

This is a behavior change, and a silently breaking one that there is no way to work around. I'm not interested in making actual behavior changes as part of this series of PRs. The fact is that it can return `null`, and that's what it needs to be annotated as.

Reply:
Is there a known use case where this would be a _real_ breaking change? I am pretty sure this field is not accessed anywhere in the roslyn-analyzers repo when the operation is obtained from the CFG. I agree that adding these null-forgiving operators as all use sites is going to be extremely confusing and burdensome.

Reply:
>This is a behavior change, and a silently breaking one that there is no way to work around. I'm not interested in making actual behavior changes as part of this series of PRs. The fact is that it can return null, and that's what it needs to be annotated as.

@333fred Then we'll have to use forgiving operator on ALL actual valid usages. While we don't even use this property when it's always null.

Reply:
If we want to consider a breaking API change, we can do that in the future. I'm specifically not making any intentional behavior changes as part of the series of PRs here.

Reply:
Agree with @333fred here. We need to be careful about separating nullable annotation passes (essentially documenting the existing behavior of our APIs) with changes that break those behaviors to better fit user scenarios. Small changes sure we make during the nullable annotation passes. But customer visible impactful changes should be considered as a separate change entirely so we can evaluate it on its individual merits